src/hypothesis/stateful.py

```diff
@@ -518,3 +575,8 @@ def execute_step(self, step):
             )
             for target in rule.targets:
                 self.bundle(target).append(VarReference(name))
+
```

Comment:
A thing that might make sense to do instead: Add a check_invariants() method to the GenericStateMachine protocol and just have the runner call that at the appropriate points rather than trying to add it to execute\_step. Then implement check_invariants() in RuleBasedStateMachine by just extracting this code into a method.